dj/construction/build.py

```diff
@@ -0,0 +1,275 @@
+"""Functions to add to an ast DJ node queries"""
+
+from functools import reduce
+from typing import Dict, List, Optional, Set, Tuple
+
+from sqlmodel import Session
+
+from dj.construction.build_planning import (
+    BuildPlan,
+    generate_build_plan_from_query,
+    optimize_level_by_cost,
+    optimize_level_by_database_id,
+)
+from dj.construction.utils import amenable_name
+from dj.errors import DJException
+from dj.models.column import Column
+from dj.models.database import Database
+from dj.models.node import Node, NodeType
+from dj.sql.parsing import ast
+from dj.sql.parsing.backends.sqloxide import parse
+
+
+# flake8: noqa: C901
+def build_select(  # pylint: disable=too-many-arguments,too-many-locals,too-many-nested-blocks,too-many-branches
```

Comment:
To piggy-back off of what the linter is suggesting, might be nice to eventually refactor into a set of smaller functions that it calls, for better readability.

Reply:
Already working on a follow up to this PR and I will refactor this.